addons/website_sale_stock/tests/test_website_sale_stock_product_warehouse.py

```diff
@@ -62,31 +62,37 @@ def test_01_get_combination_info(self):
         to the warehouse which is set in current website.
           - Set Warehouse 1, Warehouse 2 or none in website and:
             - Check available quantity of Product A and Product B in website
-        When the user doesn't set any warehouse, the module should still select
-        a default one.
+        When the user doesn't set any warehouse, the module should take into account availability in
+        all warehouses.
         """
         test_env = self.env['base'].with_context(
             website_id=self.website.id,
             website_sale_stock_get_quantity=True,
         ).env
 
-        for wh, qty_a, qty_b in [(self.warehouse_1, 10, 0), (self.warehouse_2, 15, 10), (False, 10, 0)]:
+        for wh, qty_a, qty_b in [
```

Comment:
I know you didn't write this, but putting control statements in tests is bad practice IMO. Could you split this into 3 separate tests? (it's cleaner, less error-prone, and more readable) Here's an interesting read if you're interested: https://medium.com/@kaanfurkanc/unit-testing-best-practices-3a8b0ddd88b5

Reply:
I will make 2 separate, as the first 2 cases are the same